### PR TITLE
Firestore: Add test that verifies aggregate query error message when missing index

### PR DESCRIPTION
**Pull request by dconeybe**
This verifies that the descriptive error message that occurs when the server does not have the required index has the URL that customers can follow to create the index.

This is a port of the "getCountFromServer error message is good if missing index" integration test from https://github.com/firebase/firebase-js-sdk/blob/42bfb0b8243a6c36b66a0bbedf61436e70591062/packages/firestore/test/integration/api/aggregation.test.ts#L130-L155 and https://github.com/firebase/firebase-android-sdk/pull/4232.

Googlers see b/254063570 for details.

#no-changelog

### CODE REVIEW

File: Firestore/Example/Tests/Integration/API/FIRAggregateTests.mm [hunk 1]
```diff
@@ -1093,4 +1093,45 @@ - (void)testPerformsAverageOnlyOnNumericFields {
                  [[NSNumber numberWithDouble:5] doubleValue]);
 }
 
+- (void)testFailWithGoodMessageIfMissingIndex {
```
Comment (milaGGL) [inline Firestore/Example/Tests/Integration/API/FIRAggregateTests.mm:1096]: "GoodMessage" is quite strange as it did not say what is a good or bad message. Maybe change to sth like "testFailsWithLinkToConsoleMessageIfMissingIndex"

Comment (dconeybe) [reply Firestore/Example/Tests/Integration/API/FIRAggregateTests.mm]: Done.

**Review by milaGGL — APPROVED** [review]
LGTM!

**Review by milaGGL — APPROVED** [review]
LGTM

**Pull request merged**